packages/embark-deployment/src/index.js

```diff
@@ -66,12 +66,26 @@ class Deployment {
     Object.values(contracts).forEach((contract) => {
       function deploy(result, callback) {
         if (typeof result === 'function') callback = result;
-        self.deployContract(contract, (err) => {
-          if (err) {
-            errors.push(err);
-          }
-          callback();
-        });
+        if (contract.addressHandler) {
+          self.events.request('deployment:contract:address', {contract}, (err, address) => {
+            if (err) {
+              errors.push(err);
+            } else {
+              contract.address = address;
+              contract.deployedAddress = address;
+              self.logger.info(__('{{contractName}} already deployed at {{address}}', {contractName: contract.className.bold.cyan, address: contract.address.bold.cyan}));
+              self.events.emit("deployment:contract:deployed", contract);
+            }
+            callback();
+          });
+        } else {
```

Comment:
Usual, you can return above next to the `request` and get rid of the `else`

Reply:
Yeap. Will change. The reason I do it the other way around is because I often find that an early return within an `if()` block isn't always obvious. Having `if/else` makes it very clear that it's one or the other path (as opposed to being implied).